src/fixed_rational.jl

```diff
@@ -47,14 +60,39 @@ Base.convert(::Type{F}, x::Rational) where {F<:FixedRational} = F(x)
 Base.convert(::Type{Rational{R}}, x::F) where {R,F<:FixedRational} = Rational{R}(x.num, denom(F))
 Base.convert(::Type{Rational}, x::F) where {F<:FixedRational} = Rational{eltype(F)}(x.num, denom(F))
 Base.convert(::Type{AF}, x::F) where {AF<:AbstractFloat,F<:FixedRational} = convert(AF, x.num) / convert(AF, denom(F))
-Base.convert(::Type{I}, x::F) where {I<:Integer,F<:FixedRational} = convert(I, convert(Rational, x))
+Base.convert(::Type{I}, x::F) where {I<:Integer,F<:FixedRational} =
+    let
+        isinteger(x) || throw(InexactError(:convert, I, x))
+        convert(I, div(x.num, denom(F)))
+    end
 Base.round(::Type{T}, x::F) where {T,F<:FixedRational} = div(convert(T, x.num), convert(T, denom(F)), RoundNearest)
+Base.decompose(x::F) where {T,F<:FixedRational{T}} = (x.num, zero(T), denom(F))
+
+# Promotion rules:
+function Base.promote_rule(::Type{<:FixedRational{T1,den1}}, ::Type{<:FixedRational{T2,den2}}) where {T1,T2,den1,den2}
+    return error("Refusing to promote `FixedRational` types with mixed denominators. Use `Rational` instead.")
+end
+function Base.promote_rule(::Type{<:FixedRational{T1,den}}, ::Type{<:FixedRational{T2,den}}) where {T1,T2,den}
+    return FixedRational{promote_type(T1,T2),den}
+end
+function Base.promote_rule(::Type{<:FixedRational{T1}}, ::Type{Rational{T2}}) where {T1,T2}
+    return Rational{promote_type(T1,T2)}
+end
+function Base.promote_rule(::Type{<:FixedRational{T1}}, ::Type{T2}) where {T1,T2}
+    return promote_type(Rational{T1}, T2)
+end
+
+# Want to consume integers:
 Base.promote(x::Integer, y::F) where {F<:FixedRational} = (F(x), y)
 Base.promote(x::F, y::Integer) where {F<:FixedRational} = reverse(promote(y, x))
```

Comment:
My one comment is that these should probably be turned into `promote_rule` at some point. But that's for another PR.

Also I realize this specific and unusual promotion requirement might make things a bit harder to move `FixedRational` into Ratios.jl (which I still think is a good idea, cc @oscardssmith)... But we can just figure that out later.